src/sonic-config-engine/portconfig.py

```diff
@@ -51,4 +121,115 @@ def parse_port_config_file(port_config_file):
             port_alias_map[data['alias']] = name
     return (ports, port_alias_map)
 
+# Generate configs (i.e. alias, lanes, speed, index) for port
+def gen_port_config(ports, intf, match_list, index, alias_at_lanes, lanes):
```

Comment:
As discussed, lets write a more primitive function which can be used by symmetric and asymmetric modes

Reply:
Thank you for the idea. 👍 modified to one logic.